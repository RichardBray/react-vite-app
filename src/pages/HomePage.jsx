const HomePage = () => {
  return (
    <>
      <h1>Home</h1>
      <p>
        Welcome to our video streaming website! We offer a vast collection of movies and TV shows from different genres,
        including action, comedy, romance, and science fiction. Browse through our top picks, discover new titles, and
        binge-watch your favorites. Our streaming service is available across multiple platforms, including mobile
        devices, smart TVs, and gaming consoles. We also provide high-quality video and audio to enhance your viewing
        experience. Join now and start your entertainment journey with us!
      </p>
    </>
  );
};

export default HomePage;
