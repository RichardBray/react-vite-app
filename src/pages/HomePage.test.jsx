import HomePage from './HomePage';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('HomePage', () => {
  let user;
  let inputElement;

  beforeEach(() => {
    render(<HomePage />);
    user = userEvent.setup();
    inputElement = screen.getByRole('textbox', { name: 'New task' });
  });

  afterEach(() => {
    window.localStorage.clear();
  });

  test('renders the title', () => {
    const title = screen.getByRole('heading', { name: 'Todo List' });
    expect(title).toBeInTheDocument();
  });

  test('renders the input', () => {
    expect(inputElement).toBeInTheDocument();
  });

  test('allows us to add a todo item', async () => {
    await user.type(inputElement, 'buy milk');
    await user.type(inputElement, '{Enter}');
    const todoItem = screen.getByRole('listitem', { text: 'buy milk' });
    expect(todoItem).toBeInTheDocument();
  });

  test('allows us to delete a todo item', async () => {
    await user.type(inputElement, 'buy choc');
    await user.type(inputElement, '{Enter}');

    await user.type(inputElement, 'buy magazine');
    await user.type(inputElement, '{Enter}');

    const deleteButtons = screen.getAllByRole('button', { name: 'Delete todo' });
    await user.click(deleteButtons[0]);

    const todoItems = screen.queryAllByRole('listitem');
    expect(todoItems).toHaveLength(1);
  });

  test('renders page correctly', () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('allows us to check a todo item', async () => {
    await user.type(inputElement, 'buy dinner');
    await user.type(inputElement, '{Enter}');

    const checkButton = screen.getByRole('button', { name: 'Check todo' });
    await user.click(checkButton);

    const todoItem = screen.queryByRole('listitem');

    expect(todoItem).toHaveClass('checked');
  });
});
