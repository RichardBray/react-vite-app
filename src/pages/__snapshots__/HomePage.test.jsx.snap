// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > renders page correctly 1`] = `
<DocumentFragment>
  <h1>
    Todo List
  </h1>
  <form>
    <section>
      <label
        for="todoTask"
      >
        <strong>
          New task
        </strong>
      </label>
      <br />
      <input
        id="todoTask"
        name="todo"
        placeholder="buy chocolate"
        value=""
      />
    </section>
  </form>
  <section>
    <div>
      Add a todo ðŸ‘†
    </div>
  </section>
</DocumentFragment>
`;
